cni/network/invoker_cns.go

```diff
@@ -318,16 +320,26 @@ func (invoker *CNSIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkConf
 
 			if err = invoker.cnsClient.ReleaseIPAddress(context.TODO(), ipConfig); err != nil {
 				// if the old API fails as well then we just return the error
+
 				logger.Error("Failed to release IP address from CNS using ReleaseIPAddress ",
 					zap.String("infracontainerid", ipConfigs.InfraContainerID),
 					zap.Error(err))
+
 				return errors.Wrap(err, fmt.Sprintf("failed to release IP %v using ReleaseIPAddress with err ", ipConfig.DesiredIPAddress)+"%w")
 			}
 		} else {
-			logger.Error("Failed to release IP address",
-				zap.String("infracontainerid", ipConfigs.InfraContainerID),
-				zap.Error(err))
-			return errors.Wrap(err, fmt.Sprintf("failed to release IP %v using ReleaseIPs with err ", ipConfigs.DesiredIPAddresses)+"%w")
+			var connectionErr *cnscli.ConnectionFailureErr
+			if errors.As(err, &connectionErr) {
+				addErr := fsnotify.AddFile(ipConfigs.PodInterfaceID, args.ContainerID, watcherPath)
+				if addErr != nil {
+					logger.Error("Failed to add file to watcher", zap.String("podInterfaceID", ipConfigs.PodInterfaceID), zap.String("containerID", args.ContainerID), zap.Error(addErr))
```

Comment:
return error and repeat if you failed ot add file, otherwise it will lead to a leak